### PR TITLE
Add finalize bash script, and improved build system

### PR DESCRIPTION
**Pull request by aweeks**

### CODE REVIEW

File: lib/finalize_rescue.bash [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+rescueHash=${3}
+
+cdIsoFile=`mktemp`
+
+base64 --decode | gzip -cd > ${cdIsoFile}
+mkdir -p ${cdMountpoint}
+mount -o loop ${cdIsoFile} ${cdMountpoint}
+coreos-cloudinit --from-configdrive=${cdMountpoint} --convert-netconf="debian"
+sed -ie "s/^\(${rescueUsername}\):[^:]*:\(.*\)/\1:${rescueHash}:\2/" /etc/shadow
+systemctl restart systemd-nedworkd.service
```
Comment (jayofdoom) [inline lib/finalize_rescue.bash]: s/ned/net/

File: Makefile [hunk 2]
```diff
@@ -1,2 +1,14 @@
...
+CONTAINER_UUID := $(shell uuidgen)
+
+default: docker
+
+docker:
+	mkdir -p ./build
+	docker build --no-cache -t "$(IMAGE_UUID)" .
+	docker run --name="$(CONTAINER_UUID)" "$(IMAGE_UUID)" /bin/bash
+	docker cp $(CONTAINER_UUID):/usr/local/bin/onmetal-rescue-agent ./build
+	docker cp $(CONTAINER_UUID):/usr/local/bin/finalize_rescue.bash ./build
```
Comment (jimrollenhagen) [inline Makefile]: Should we remove the image when this is done with?

Comment (jimrollenhagen) [inline Makefile]: Or even in `make clean`?

**Comment by jimrollenhagen** [comment]
Looks mostly good, one thing inline. Would love eyes on this other than mine as well.

File: Makefile (hunk 2 above)
Comment (aweeks) [inline Makefile]: done

**Comment by pcsforeducation** [comment]
LGTM

**Pull request merged**